scrappers/tbib/scraper.rb

```diff
@@ -0,0 +1,94 @@
+raise 'missing gems.txt!' unless File.exist? 'gems.txt'
+File.open('gems.txt', 'r').read.split("\n").each { |x| require x }
+Dir['lib/*.rb'].each { |x| require_relative x }
+$log = Logger.new
+
+Config = Struct.new(:server_ip, :server_port, :save_path, :name, :database, :table, :tags) do
+	def info
+		ret = "\n"
+		[[server_port, 'Server Port'],
+		[server_ip, 'Server IP'],
+		[save_path, 'Save Path'],
+		[name, 'Name'],
+		[database, 'Database'],
+		[table, 'Table'],
+		[tags, 'Tags']].each do |t|
+			a = t[0]
+			b = t[1]
+			ret << " * #{b}: #{a.is_a?(Array) ? a.to_s : a}\n"
+		end
+		ret
+	end
+
+	def verify!
+		marr = []
+		[[server_port, 'Server Port'],
+		[server_ip, 'Server IP'],
+		[save_path, 'Save Path'],
+		[name, 'Name'],
+		[database, 'Database'],
+		[table, 'Table'],
+		[tags, 'Tags']].each do |t|
+			if t[0].nil?
+				marr << "Value '#{t[1]}' is missing from the configuration file!"
+			end
+		end
+		marr.each { |x| $log.fatal(x) }
+		exit 1 if marr.length > 0
+	end
+end
+
+$config = Config.new nil, nil, nil, nil, nil
+
+has_config = false
+
+OptionParser.new do |ops|
+	ops.banner = 'Usage: ruby scraper.rb [options]'
+
+	ops.on('-h', '--help', 'Prints this text.') do
+		puts ops
+		exit 0
+	end
+
+	ops.on('-d', '--default-config', 'Makes the scraper use the default configuration options.') do
```

Comment:
This should generate and dump the default `config.toml` to the stdout, not make the scrapper use the default config